src/main/java/core/basesyntax/service/Parser.java

```diff
@@ -0,0 +1,7 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.TransactionDto;
+
+public interface Parser<T> {
+    TransactionDto parseTo(String line);
```

Comment:
ты не используешь параметризацию
```suggestion
    T parseTo(String line);
```